### PR TITLE
chore(deps): upgrade decode-uri-component to >= 0.2.2

### LINKED ISSUE #2243: chore(deps): upgrade decode-uri-component to  >= 0.2.2
### Description

The robot sent in a PR [1] to force the the transitive dependency upgrades via the lockfile without touching our actual dependencies, but we should follow-up to that by upgrading the direct dependencies as well so that in the event of the lock file getting wiped and recreated (which sometimes is necessary/hard to avoid) we won't fall back into having these vulnerabilities again.

[1] https://github.com/hyperledger/cactus/pull/2226

```sh
$ npm ls  decode-uri-component
@hyperledger/cactus@ /..../cactus
├─┬ @hyperledger/cactus-cmd-socketio-server@1.1.3 -> ./packages/cactus-cmd-socketio-server
│ └─┬ web3@1.6.0
│   └─┬ web3-bzz@1.6.0
│     └─┬ swarm-js@0.1.40
│       └─┬ xhr-request@1.1.0
│         └─┬ query-string@5.1.1
│           └── decode-uri-component@0.2.0 deduped
├─┬ @hyperledger/cactus-example-carbon-accounting-frontend@1.1.3 -> ./examples/cactus-example-carbon-accounting-frontend
│ └─┬ @angular-devkit/build-angular@13.3.2
│   └─┬ stylus@0.56.0
│     └─┬ css@3.0.0
│       └─┬ source-map-resolve@0.6.0
│         └── decode-uri-component@0.2.0 deduped
├─┬ @hyperledger/cactus-test-tooling@1.1.3 -> ./packages/cactus-test-tooling
│ └─┬ keycloak-admin@1.14.21
│   └─┬ query-string@6.14.1
│     └── decode-uri-component@0.2.0
└─┬ @hyperledger/cactus-verifier-client@1.1.3 -> ./packages/cactus-verifier-client
  └─┬ jest-extended@0.11.5
    └─┬ expect@24.9.0
      └─┬ jest-message-util@24.9.0
        └─┬ micromatch@3.1.10
          └─┬ snapdragon@0.8.2
            └─┬ source-map-resolve@0.5.3
              └── decode-uri-component@0.2.0 deduped
```

### PR DESCRIPTION
**Pull request by charellesandig**
Fixes #2243 

Depends on: https://github.com/hyperledger/cacti/issues/2357

Signed-off-by: charellesandig charelle.wrk@gmail.com

### CODE REVIEW

**Review by jagpreetsinghsasan — APPROVED** [review]
LGTM

**Review by petermetz — CHANGES_REQUESTED** [review]
@charellesandig Please remove all occurrences of the dependency not just one.

```sh
$ npm ls decode-uri-component
@hyperledger/cactus@ /home/peter/a/blockchain/cactus
├─┬ @hyperledger/cactus-cmd-socketio-server@1.1.3 -> ./packages/cactus-cmd-socketio-server
│ └─┬ web3@1.8.2
│   └─┬ web3-bzz@1.8.2
│     └─┬ swarm-js@0.1.40
│       └─┬ xhr-request@1.1.0
│         └─┬ query-string@5.1.1
│           └── decode-uri-component@0.2.2 deduped
├─┬ @hyperledger/cactus-example-supply-chain-frontend@1.1.3 -> ./examples/cactus-example-supply-chain-frontend
│ └─┬ @angular-devkit/build-angular@13.3.5
│   └─┬ stylus@0.56.0
│     └─┬ css@3.0.0
│       └─┬ source-map-resolve@0.6.0
│         └── decode-uri-component@0.2.2 deduped
└─┬ @hyperledger/cactus-test-tooling@1.1.3 -> ./packages/cactus-test-tooling
  └─┬ keycloak-admin@1.14.22
    └─┬ query-string@6.14.1
      └── decode-uri-component@0.2.2

```

**Comment by charellesandig** [comment]
Hi @petermetz ,  for these packages I upgraded the direct dependencies where `decode-uri-component` occurs.

```
packages/cactus-cmd-socketio-server
packages/cactus-cmd-socketio-server
cactus-test-tooling
cactus-verifier-client
```

For `cactus-verifier-client` when I upgraded from "jest-extended": "0.11.5" to "jest-extended": "3.0.1" the `docode-uri-component` is not present anymore.

File: examples/cactus-example-carbon-accounting-frontend/package.json
Comment (charellesandig) [reply examples/cactus-example-carbon-accounting-frontend/package.json:74]: Hi @petermetz , running `yarn build:dev` on this branch is not giving successful results. I also tried running it without my changes and it still fails. The error comes from `examples/cactus-workshop-examples-2022-11-14`.

Comment (charellesandig) [reply examples/cactus-example-carbon-accounting-frontend/package.json:74]: Noted on this @petermetz , I will monitor and run test once it is resolved. Thank you!

**Comment by charellesandig** [comment]
Depends on: #2357

**Review by petermetz — APPROVED** [review]
LGTM

**Review by petermetz — CHANGES_REQUESTED** [review]
Can't merge until maintainer consensus has been achieved.

**Review by izuru0 — APPROVED** [review]

**Review by petermetz — APPROVED** [review]
I rebased onto upstream/main, fixed the merge conflicts and in the meantime Izuru reviewed it too, so we are good to go.

**Pull request merged**
